tutorial_import_products_inline.md

```diff
@@ -0,0 +1,95 @@
+# **Import Products From Inline Source Tutorial**
+
+## Let's get started
+
+The inline importing is a convenient way to make bulk changes in a catalog:
+
+- import some amount of products, no more than 100 at a time,
+- update some products set,
+- make fast and frequent changes of products quantity or price or any other field.
+
+To import your products into a catalog inline you should prepare the ```product_inline_source``` object which is a set
+of products. Each product should be provided in a JSON format as single line (one product per line with line breaks as a
+delimiter). An example of a product in JSON format can be found in
+a [Retail API documentation](https://cloud.google.com/retail/docs/upload-catalog#json-format)
+
+More information about different import types, their limitations and use cases you can find
+in [Retail API documentation](https://cloud.google.com/retail/docs/upload-catalog#considerations)
+
+**Time to complete**: About 2 minutes
+
+## Before you begin
+
+To run Python code samples from this tutorial you will need to set up your virtual environment.
+
+To do that, run the following commands in a terminal:
+
+```bash
+pip install virtualenv
+```
+
+```bash
+virtualenv <your-env>
+```
+
+```bash
+source <your-env>/bin/activate
+```
+
+Next, install Google packages:
+
+```bash
+pip install google
+```
+
+```bash
+pip install google-cloud-retail
+```
+
+**Tip**: Click the copy button on the side of the code box to later paste the command in the Cloud Shell terminal and
+run it.
+
+## Import products from inline source
+
+It is worth to mention that for inline import the only INCREMENTAL reconciliation mode is available, it means that
+during the importing new products will be inserted and existing products will be updated, if there are products in
+catalog which are absent in the import source JSON they will remain unchanged.
+
+Please open **product/import_products_inline_source.py** and look at the example of import product request.
+
+The field ```parent``` contains a **catalog name** along with a branch number to which you are going to import your
```

Comment:
The field ```parent``` contains a **catalog name** along with a branch number you are going to import your
 products to.